vendor/bat-native-ledger/src/bat/ledger/internal/core/bat_ledger_context_unittest.cc

```diff
@@ -7,78 +7,45 @@
 
 #include <string>
 
-#include "bat/ledger/internal/core/async_result.h"
 #include "bat/ledger/internal/core/bat_ledger_test.h"
+#include "bat/ledger/internal/core/future.h"
 
 namespace ledger {
 
 class BATLedgerContextTest : public BATLedgerTest {};
 
-struct TestSettings {
-  std::string a;
-  int b;
-
-  static const TestSettings kProduction;
-  static const TestSettings kStaging;
-  static const TestSettings kDevelopment;
-};
-
-const TestSettings TestSettings::kProduction = {.a = "prod", .b = 1};
-const TestSettings TestSettings::kStaging = {.a = "staging", .b = 2};
-const TestSettings TestSettings::kDevelopment = {.a = "dev", .b = 3};
-
-TEST_F(BATLedgerContextTest, GetSettings) {
-  struct ResetEnv {
-    ResetEnv() : env(ledger::_environment) {}
-    ~ResetEnv() { ledger::_environment = env; }
-    mojom::Environment env;
-  } reset_env;
-
-  ASSERT_EQ(ledger::_environment, mojom::Environment::PRODUCTION);
-  ledger::_environment = mojom::Environment::PRODUCTION;
-  auto& prod = context()->GetSettings<TestSettings>();
-  ASSERT_EQ(prod.a, "prod");
-  ASSERT_EQ(prod.b, 1);
-
-  ledger::_environment = mojom::Environment::STAGING;
-  auto& staging = context()->GetSettings<TestSettings>();
-  ASSERT_EQ(staging.a, "staging");
-  ASSERT_EQ(staging.b, 2);
+class TestComponent : public BATLedgerContext::Object {
+ public:
+  static const char kContextKey[];
```

Comment:
As of C++17, this can be an inline variable:
```
inline static const char kContextKey[] = "test-component";
```
You can even put it in a header file and include it in multiple translation units, without getting a "multiply defined symbols" linker error.

Reply:
❤️